bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/ComfoAirHandler.java

```diff
@@ -0,0 +1,434 @@
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.transport.serial.SerialPortManager;
+import org.openhab.binding.comfoair.internal.datatypes.ComfoAirDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ComfoAirHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hans Böhm - Initial contribution
+ */
+@NonNullByDefault
+public class ComfoAirHandler extends BaseThingHandler {
+    private static final int DEFAULT_REFRESH_INTERVAL_SEC = 60;
+
+    private final Logger logger = LoggerFactory.getLogger(ComfoAirHandler.class);
+    private final ComfoAirConfiguration config = getConfigAs(ComfoAirConfiguration.class);
+    private final SerialPortManager serialPortManager;
+    private @Nullable ScheduledFuture<?> poller;
+    private @Nullable ScheduledFuture<?> affectedItemsPoller;
+    private @Nullable ComfoAirSerialConnector comfoAirConnector;
+
+    public static final int BAUDRATE = 9600;
+
+    public ComfoAirHandler(Thing thing, final SerialPortManager serialPortManager) {
+        super(thing);
+        this.serialPortManager = serialPortManager;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channelId = channelUID.getId();
+
+        if (command instanceof RefreshType) {
+            Channel channel = this.thing.getChannel(channelUID);
+            if (channel != null) {
+                updateChannelState(channel);
+            }
+        } else {
+            ComfoAirCommand changeCommand = ComfoAirCommandType.getChangeCommand(channelId, command);
+
+            if (changeCommand != null) {
+                Set<String> keysToUpdate = getThing().getChannels().stream().map(Channel::getUID).filter(this::isLinked)
+                        .map(ChannelUID::getId).collect(Collectors.toSet());
+                sendCommand(changeCommand, channelId);
+
+                Collection<ComfoAirCommand> affectedReadCommands = ComfoAirCommandType
+                        .getAffectedReadCommands(channelId, keysToUpdate);
+
+                if (affectedReadCommands.size() > 0) {
+                    Runnable updateThread = new AffectedItemsUpdateThread(affectedReadCommands);
+                    affectedItemsPoller = scheduler.schedule(updateThread, 3, TimeUnit.SECONDS);
+                }
+            } else {
+                logger.warn("Unhandled command type: {}, channelId: {}", command.toString(), channelId);
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        String serialPort = this.config.serialPort;
+
+        if (serialPort.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
```

Comment:
You might want to give the user a hint what went wrong by passing a message as a third argument.